Issue/138/xgal pz tutorial

sschmidt23 opened the pull request:
This PR addresses issue #138, I've updated the tutorial cells based on feedback from two people who I asked to work through the tutorial, and I am now happy with the tutorial.  

There is still a sys.path.insert to load the newest version of the reader, but that single cell can be removed once gar-catalogs v0.19.0 is propagated to the desc-python kernel.

yymao commented on tutorials/extragalactic_gcr_photoz_catalogs.ipynb:
```diff
@@ -0,0 +1,690 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#NOTE: can't grab an @property from the linked catalog in a composite catalog!  Grab from the original catalog instead\n",
+    "xcat = GCRCatalogs.load_catalog(\"photoz_pdfs_v1_for_cosmoDC2_v1.1.4_image\")\n",
+    "zgrid = xcat.photoz_pdf_bin_centers"
```
@sschmidt23 I realize the reason that you cannot access add-on attributes is that `only_use_master_attr` was set to True in the photo-z composite configs. In the future we should set it to False. For now we can edit it during runtime.

```suggestion
    "# NOTE: Need to set `only_use_master_attr` to False to allow accessing add-on attributes\n",
    "cat.only_use_master_attr = False\n",
    "zgrid = cat.photoz_pdf_bin_centers"
```

sschmidt23 replied:
done

yymao commented on tutorials/extragalactic_gcr_photoz_catalogs.ipynb:
```diff
@@ -0,0 +1,690 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xcat2 = GCRCatalogs.load_catalog(\"photoz_fzboost_for_cosmoDC2_v1.1.4_image\")\n",
+    "fz_zgrid = xcat2.photoz_pdf_bin_centers"
```
Same as above
```suggestion
    "cat2.only_use_master_attr = False  # allow accessing add-on attributes\n",
    "fz_zgrid = cat2.photoz_pdf_bin_centers"
```

sschmidt23 replied:
done

yymao commented on tutorials/extragalactic_gcr_photoz_catalogs.ipynb:
```diff
@@ -0,0 +1,690 @@
+    "if you want to follow along but do not wish to actually run the notebook live at NERSC."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Temporary path to gcr-catalogs until new version is included in desc-python kernel:\n",
+    "import sys\n",
+    "sys.path.insert(0,\"/global/cfs/cdirs/lsst/groups/PZ/Packages/gcr-catalogs/lib/python3.7/site-packages/GCRCatalogs-0.18.1-py3.7.egg\")"
+   ]
+  },
```
Remove this cell when the kernel is updated.
```suggestion
```

sschmidt23 replied:
removed

sschmidt23 commented:
Ah, I should have noticed that `use_only_master_attr`, thanks for pointing this out!

I'll wait until desc-python is updated to v0.19.0, check that everything works, then delete the path before merging just to be sure it all works smoothly.

sschmidt23 commented:
Ok, I added only_use_master_attr =False in the two locations and removed the local loading of gcr-catalogs, everything works now that desc-python points to v0.19.0.  Everything looks ready to go.
